drivers/ethernet/phy/phy_adin2111.c

```diff
@@ -455,6 +455,28 @@ static int phy_adin2111_init(const struct device *dev)
 	return phy_adin2111_sft_pd(dev, false);
 }
 
+int phy_adin2111_mdio_read_c22(const struct device *dev, int reg_addr, uint16_t *val)
+{
+	return phy_adin2111_c22_read(dev, reg_addr, val);
+}
+
+int phy_adin2111_mdio_write_c22(const struct device *dev, int reg_addr, uint16_t val)
+{
+	return phy_adin2111_c22_write(dev, reg_addr, val);
+}
+
+int phy_adin2111_mdio_read_c45(const struct device *dev, uint16_t devad,
+			       uint16_t reg, uint16_t *val)
+{
+	return phy_adin2111_c45_read(dev, devad, reg, val);
+}
+
+int phy_adin2111_mdio_write_c45(const struct device *dev, uint16_t devad,
+				uint16_t reg, uint16_t val)
+{
+	return phy_adin2111_c45_write(dev, devad, reg, val);
+}
```

Comment:
What is the goal here? There is [adin mdio driver](https://github.com/zephyrproject-rtos/zephyr/blob/be881d4cf269d3126722ce42e3c198b5960a3f3b/drivers/mdio/mdio_adin2111.c).

Reply:
@GeorgeCGV  I see mdio driver works fine, but due to the big phy register set, idea is to have such c22/c45 functions as APIs too, and only way i found to have them accessible is by providing a *phy reference from adin2111 API and accessing them as phy APIs. Maybe the name should be phy_adin2111_write_c45() (without mdio).

Reply:
You can try to get correctio MDIO like `mdio_shell.c` does:
`const struct device *mdio = DEVICE_DT_GET(DT_COMAT_GET_ANY_STATUS_OKAY(adi_adin2111_mdio));`

Then, using the MDIO API RW for both clauses is possible.

Reply:
Thanks a lot, used mdio_read() and realted APIs.